### ISSUE TITLE
Duplicate nullable warnings are reported for conversion from a base class to derived

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#30941]
```
        [Fact]
        public void Test()
        {
            var source = @"
class Outer : Base
{
    void M1(Base? x1)
    {
        Outer y = x1;
    }
}

class Base {}
";

            CreateCompilation(source, options: WithNonNullTypesTrue()).VerifyDiagnostics(
                // (6,19): error CS0266: Cannot implicitly convert type 'Base' to 'Outer'. An explicit conversion exists (are you missing a cast?)
                //         Outer y = x1;
                Diagnostic(ErrorCode.ERR_NoImplicitConvCast, "x1").WithArguments("Base", "Outer").WithLocation(6, 19),
                // (6,19): warning CS8600: Converting null literal or possible null value to non-nullable type.
                //         Outer y = x1;
                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "x1").WithLocation(6, 19),
                // (6,19): warning CS8619: Nullability of reference types in value of type 'Base' doesn't match target type 'Outer'.
                //         Outer y = x1;
                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "x1").WithArguments("Base", "Outer").WithLocation(6, 19)
                );
        }
```

It feels like the WRN_ConvertingNullableToNonNullable and WRN_NullabilityMismatchInAssignment warnings are complaining about the same thing.

**Comment by jcouv** [comment]
This appears to be fixed. I'll add a test to close
